server/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorServiceTests.java

```diff
@@ -0,0 +1,301 @@
+        }
+
+        var nodesShutdownMetadata = new NodesShutdownMetadata(
+            nodeShutdowns.stream()
+                .collect(
+                    toMap(
+                        it -> it.nodeId,
+                        it -> SingleNodeShutdownMetadata.builder()
+                            .setNodeId(it.nodeId)
+                            .setType(it.type)
+                            .setReason("test")
+                            .setNodeSeen(true)
+                            .setStartedAtMillis(System.currentTimeMillis())
+                            .setAllocationDelay(
+                                it.allocationDelaySeconds != null ? TimeValue.timeValueSeconds(it.allocationDelaySeconds) : null
+                            )
+                            .build()
+                    )
+                )
+        );
+
+        return ClusterState.builder(new ClusterName("test-cluster"))
+            .routingTable(builder.build())
+            .metadata(Metadata.builder().putCustom(NodesShutdownMetadata.TYPE, nodesShutdownMetadata).build())
+            .build();
+    }
+
+    private static Map<String, Object> addDefaults(Map<String, Object> override) {
+        return Map.of(
+            "unassigned_primaries",
+            override.getOrDefault("unassigned_primaries", 0),
+            "initializing_primaries",
+            override.getOrDefault("initializing_primaries", 0),
+            "restarting_primaries",
+            override.getOrDefault("restarting_primaries", 0),
+            "started_primaries",
+            override.getOrDefault("started_primaries", 0),
+            "unassigned_replicas",
+            override.getOrDefault("unassigned_replicas", 0),
+            "initializing_replicas",
+            override.getOrDefault("initializing_replicas", 0),
+            "restarting_replicas",
+            override.getOrDefault("restarting_replicas", 0),
+            "started_replicas",
+            override.getOrDefault("started_replicas", 0)
+        );
+    }
+
+    private static IndexRoutingTable index(String name, ShardAllocation primaryState, ShardAllocation... replicaStates) {
+        var index = new Index(name, UUID.randomUUID().toString());
+        var shardId = new ShardId(index, 1);
+
+        var builder = IndexRoutingTable.builder(index);
+        builder.addShard(createShardRouting(shardId, true, primaryState));
+        for (var replicaState : replicaStates) {
+            builder.addShard(createShardRouting(shardId, false, replicaState));
+        }
+        return builder.build();
+    }
+
+    private static ShardRouting createShardRouting(ShardId shardId, boolean primary, ShardAllocation allocation) {
+        var routing = newUnassigned(
+            shardId,
+            primary,
+            primary ? RecoverySource.EmptyStoreRecoverySource.INSTANCE : RecoverySource.PeerRecoverySource.INSTANCE,
+            new UnassignedInfo(UnassignedInfo.Reason.INDEX_CREATED, null)
```

Comment:
I suppose we do not trigger on this, but neither do we on recovery. I wonder if we should randomize between more causes here, like `NODE_LEFT`, `PRIMARY_FAILED` and more?

Reply:
May be. Due to the way how shards are constructed this could be done by adding more `ShardState` and extending this method with additional simulation